src/Functions/EntropyLeardnedHash.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+// Reference to the paper about entropy-learned hashing:
+// https://bhentsch.github.io/doc/EntropyLearnedHashing.pdf
+
+#include <cstdint>
+#include <limits>
+#include <map>
+#include <string>
+#include <vector>
+
+// Key is a byte string
+namespace EntropyLearnedHashing
+{
+using Key = std::string;
+
+[[nodiscard]] Key getPartialKey(const std::string & key, const std::vector<size_t> & positions);
+
+class IDsManager
+{
+private:
+    static IDsManager * instance_ptr;
+    IDsManager() { }
+    IDsManager(const IDsManager &);
+    IDsManager & operator=(IDsManager &);
+
+public:
+    static IDsManager * getInstance()
+    {
+        if (!IDsManager::instance_ptr)
+            IDsManager::instance_ptr = new IDsManager();
+        return IDsManager::instance_ptr;
+    }
+
+    using IDType = std::string;
+    std::map<IDType, std::vector<size_t>> positions_by_id;
+    static IDType default_id;
```

Comment:
Can't find where this is defined.

Reply:
I thought to have `default_id` for a function which takes a single key, described here: https://github.com/ClickHouse/ClickHouse/pull/48021#discussion_r1181313446. If we don't need such a function, I will delete this field of singleton.